docs/spec_draft.md

```diff
@@ -604,6 +605,86 @@ Afterwards, within each `process_group`:
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.all_reduce
+
+### Semantics
+
+Within each process group in the StableHLO grid, performs reduction, using
+`computations`, over the  values of the `operand` tensor from each process and
+produces a `result` tensor.
```

Comment:
Proposed rewording:

```
Within each process group in the StableHLO grid, applies a reduction function 
`computation` to the values of the `operand` tensor from each process and
produces a `result` tensor.
```

Feedback:
  1. Reworded slightly to be consistent with how we specced `reduce` (see below for an excerpt).
  2. Typo: "computation".


```
// reduce
Applies a reduction function `body` to `inputs` and `init_values` along the
`dimensions` and produces a `result` tensor.
```